planet4-gutenberg-blocks.php

```diff
@@ -251,35 +255,12 @@ function set_allowed_block_types( $allowed_block_types, $post ) {
 		'campaign' => $campaign_block_types,
 	];
 
-	$allowed_p4_block_types = $all_allowed_p4_block_types[ $post->post_type ];
-
-	if ( empty( $allowed_p4_block_types ) ) {
-		return $wordpress_blocks;
-	}
-
-	$allowed_block_types = array_merge( $wordpress_blocks, $allowed_p4_block_types );
+	$allowed_p4_block_types = $all_allowed_p4_block_types[ $post_type ] ?? $all_allowed_p4_block_types['page'];
```

Comment:
@comzeradd This previously resulted in warnings if there's no post or no known post type.
Afaik this only occurs for "reusable blocks". Those actually can contain blocks from all post types, depending on where they are saved as a reusable block. However we default to adding no P4 blocks. So this prevented you from putting new P4 blocks in an existing reusable block.

Reply:
Yeah, that's a good fix. So far the editors had to create Reusable blocks only in the Page editor just to be able to add P4 blocks.